sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageExtensions.cs

```diff
@@ -23,15 +24,50 @@ public static AmqpMessage ToAmqpMessage(this ServiceBusMessage message)
             }
             throw new NotSupportedException($"{message.AmqpMessage.Body.GetType()} is not a supported message body type.");
         }
 
+        private static AmqpMessage CreateValueMessage(object value)
+        {
+            return value switch
+            {
+                IEnumerable<KeyValuePair<string, string>> enumerable => AmqpMessage.Create(new AmqpValue { Value = CreateAmqpMap(enumerable) }),
+                IEnumerable<KeyValuePair<string, object>> enumerable => AmqpMessage.Create(new AmqpValue { Value = CreateAmqpMap(enumerable) }),
+                IEnumerable<KeyValuePair<string, byte>> enumerable => AmqpMessage.Create(new AmqpValue { Value = CreateAmqpMap(enumerable) }),
+                IEnumerable<KeyValuePair<string, sbyte>> enumerable => AmqpMessage.Create(new AmqpValue { Value = CreateAmqpMap(enumerable) }),
+                IEnumerable<KeyValuePair<string, char>> enumerable => AmqpMessage.Create(new AmqpValue { Value = CreateAmqpMap(enumerable) }),
+                IEnumerable<KeyValuePair<string, short>> enumerable => AmqpMessage.Create(new AmqpValue { Value = CreateAmqpMap(enumerable) }),
+                IEnumerable<KeyValuePair<string, ushort>> enumerable => AmqpMessage.Create(new AmqpValue { Value = CreateAmqpMap(enumerable) }),
+                IEnumerable<KeyValuePair<string, int>> enumerable => AmqpMessage.Create(new AmqpValue { Value = CreateAmqpMap(enumerable) }),
+                IEnumerable<KeyValuePair<string, uint>> enumerable => AmqpMessage.Create(new AmqpValue { Value = CreateAmqpMap(enumerable) }),
+                IEnumerable<KeyValuePair<string, long>> enumerable => AmqpMessage.Create(new AmqpValue { Value = CreateAmqpMap(enumerable) }),
+                IEnumerable<KeyValuePair<string, ulong>> enumerable => AmqpMessage.Create(new AmqpValue { Value = CreateAmqpMap(enumerable) }),
+                IEnumerable<KeyValuePair<string, float>> enumerable => AmqpMessage.Create(new AmqpValue { Value = CreateAmqpMap(enumerable) }),
+                IEnumerable<KeyValuePair<string, double>> enumerable => AmqpMessage.Create(new AmqpValue { Value = CreateAmqpMap(enumerable) }),
+                IEnumerable<KeyValuePair<string, decimal>> enumerable => AmqpMessage.Create(new AmqpValue { Value = CreateAmqpMap(enumerable) }),
+                IEnumerable<KeyValuePair<string, bool>> enumerable => AmqpMessage.Create(new AmqpValue { Value = CreateAmqpMap(enumerable) }),
+                IEnumerable<KeyValuePair<string, Guid>> enumerable => AmqpMessage.Create(new AmqpValue { Value = CreateAmqpMap(enumerable) }),
+                IEnumerable<KeyValuePair<string, DateTime>> enumerable => AmqpMessage.Create(new AmqpValue { Value = CreateAmqpMap(enumerable) }),
+                _ => AmqpMessage.Create(new AmqpValue { Value = value }),
+            };
+        }
+
+        private static AmqpMap CreateAmqpMap<K,V>(IEnumerable<KeyValuePair<K,V>> enumerable)
+        {
+            AmqpMap map = new();
```

Comment:
Is it now allowed to use this declaration? I saw it is rarely used currently

Reply:
Yes; we're able to use C# 9 constructs now.